frontend/src/components/chat/ChatComponent.vue

```diff
@@ -220,22 +226,34 @@ onMounted(() => {
 
   //You Prometed
   socket.on("user-promoted", (eventData) => {
-    console.log("Promoted" , eventData);
-    const { channelId, message} = eventData;
+    const { channelId, message, userId} = eventData;
 
-    const curUser = getUserInChannel(channelId, userStore().user.id);
-    if (curUser)
-      curUser.isAdmin = true;
+    if (message.includes("You have been promoted in channel")) {
+      const curUser = getUserInChannel(channelId, userStore().user.id);
+      if (curUser)
+        curUser.isAdmin = true;
+    }
+    else {
+      const curUser = getUserInChannel(channelId, userId);
+      if (curUser)
+        curUser.isAdmin = true;
+    }
```

Comment:
Qual é a diferença? É para diferenciar entre o user atual e os outros? é que parece que entre usares o userId ou o userStore().user.id vai dar ao mesmo, vai meter o curUser como admin

Reply:
O userId vem do servidor, o problema é que o backend neste momento não envia o parametro userId quando é o alvo sou eu propio, para isso uma alternativa que fiz foi olhar para a menssagem, e quando o evento alvo for eu mesmo eu vou buscar o meu propio userId (userStore.id)